webservice/soap/classes/class.ilSoapUserAdministration.php

```diff
@@ -99,6 +99,71 @@ function loginLDAP($client, $username, $password)
 		return $this->login($client, $username, $password);
 	}
 
+	/**
+	 * login as user for studip mode
+	 * @param string $sid
+	 * @param int $user_id
+	 * @return string $sid
+	 */
+	public function loginAsUser($sid, $user_id)
+	{
+		global $rbacsystem, $ilUser, $ilIliasIniFile;
+
+		$this->initAuth($sid);
+		$this->initIlias();
+		list($admin_sid,$client) = $this->__explodeSid($sid);
+
+		if(!$this->__checkSession($sid))
+		{
+			return $this->__raiseError($this->__getMessage(),$this->__getMessageCode());
+		}
+
+		if (!$ilIliasIniFile->readVariable('server', 'studip'))
+		{
+			return $this->__raiseError('Stud.IP mode not active.','Server');
+		}
+
```

Comment:
Another - more restrictive approach is:

if($rbacreview->isAssigned($ilUser->getId(),SYSTEM_ROLE_ID))

Thus only soap users assigned to the global administrator role are allowed to create foreign sessions.